### DISCUSSION TITLE
Providing Additional Min/Max Temperature Readings to dbus-serialbattery

### DISCUSSION THREAD

**Discussion by BipedalPrimate** [mr-manuel/venus-os_dbus-serialbattery#88] [Q&A] [answered]
The older style Jkbms (the non-PB versions) provide 2 temperature sensors which dbus-serialbattery uses.

dbus-serialbattery has provision for 4 temperature sensors.

I plan to add more temperature sensors to the battery & hope to send the Min/Max temperature readings to dbus-serialbattery Temp 3 & Temp 4 by a method yet to be determined.

This will (hopefully) allow dbus-serialbattery to include these additional temperature readings in the Charge (CCCM_T)/Discharge (DCCM_T) algorithms.

From my initial scan, jkbms.py at 
https://github.com/mr-manuel/venus-os_dbus-serialbattery/blob/d714430b04dd001d74790b7eb084518bb14bd483/etc/dbus-serialbattery/bms/jkbms.py#L125-L137
would be the place to insert some code to achieve the desired outcome.

Is that the correct location to insert a Min & Max temperature reading to Temperature3 & Temperature4?

My thoughts are to have the additional individual temperature readings recorded in dbus somewhere (com.victronenergy.temperature?) and determine the Min/Max for insert into dbus-serialbattery.  Is that a good plan are there other alternatives?

Temperature3 & Temperature4 are already provided for in dbus SerialBattery(JKBMS):
root@raspberrypi4:~# dbus-spy
com.victronenergy.battery.ttyUSB2
System/Temperature1                                                           22
System/Temperature1Name                                                    Front
System/Temperature2                                                           23
System/Temperature2Name                                                     Rear
System/Temperature3                                                            -
System/Temperature3Name                                                   Temp 3
System/Temperature4                                                            -
System/Temperature4Name                                                   Temp 4

**Comment by mr-manuel** [comment]
Yes this is the correct place to add it. If you only want to add temperatures to Venus OS you can use dbus-mqtt-temperature.

Another solution could be to use all values from the dbus-serialbattery driver, make the modifications you need in Node-RED and populate dbus-mqtt-battery. This would allow updating the drivers without loosing your changes.

**Comment by BipedalPrimate** [comment]
As I already using VenusOS-TemperatureService with 1-wire sensors, the temperature values are automatically added to Venus OS/VRM via com.victronenergy.temperature.wire_idxx where xx is currently 02 to 05.

My intent is to allow dbus-serialbattery to use the Temperature Limitation affecting CVL/DCL based on max/min temperature values I would feed to dbus-serialbattery.

Trying to understand how the alternative solution of using dbus-mqtt-battery would allow the Temperature Limitation functionality of dbus-serialbattery to operate.  From what I read in dbus-mqtt-battery, the battery data is written to different dbus path than that which dbus-serialbattery uses.  Could you explain that solution a bit more?

Thanks.

**Reply by mr-manuel** [reply]
dbus-mqtt-battery displays only the data you provide to the driver with MQTT. You can also select it as BMS under DVCC. You could fetch all values from dbus-serialbattery, then modify CVL, CCL, DCL the way you like and feed them to dbus-mqtt-battery. This way you can use your own algorithm based on the factors you need, like your temperature sensors.

**Comment by mr-manuel** [comment, marked as answer]
A new BMS will be added that fetches data from a MQTT topic in Q3 2025.
